### PR TITLE
Savannah

### PR DESCRIPTION
**Pull request by Ars6300**
Things left to do: 
- [x] close button
- [x] sounds
- [x] end game stats

### CODE REVIEW

**Review by Diana-Valeeva — APPROVED** [review]

File: src/app/core/models/ISavannahGame.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { ITrainedWord } from './ITrainedWord';
+import { IWord } from './IWord';
+
+export enum GameState {
```
Comment (rss-maksim) [inline src/app/core/models/ISavannahGame.ts:4]: Sometimes it's enough to have just keys in enum

```
export enum GameState {
  SETTING,
  PREP,
...
}

File: src/app/mini-games/savannah/components/difficultyy-input/difficultyy-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+@Component({
+  selector: 'app-difficultyy-input',
+  templateUrl: './difficultyy-input.component.html',
+  styleUrls: ['./difficultyy-input.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DifficultyyInputComponent {
+  @Output() startWithDifficulty = new EventEmitter<number>();
+  constructor() {}
+  difficulty!: number;
```
Comment (rss-maksim) [inline src/app/mini-games/savannah/components/difficultyy-input/difficultyy-input.component.ts:12]: It's better to move up

File: src/app/mini-games/savannah/components/lifes/lifes.component.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  left: 10px;
+  display: flex;
+  flex-direction: row-reverse;
+}
+
+.icon {
+  user-select: none;
+}
+
+@media screen and (max-width: 500px) {
```
Comment (rss-maksim) [inline src/app/mini-games/savannah/components/lifes/lifes.component.scss:13]: Ideally to create variable for ```max-width```

File: src/app/mini-games/savannah/components/main-actions/main-actions.component.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  styleUrls: ['./main-actions.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MainActionsComponent {
+  @Input() word!: string;
+  @Input() translation!: string;
+  @Input() answers!: string[];
+  @Input() lifes!: number;
+  @Input() progress!: number;
+  @Output()
```
Comment (rss-maksim) [inline src/app/mini-games/savannah/components/main-actions/main-actions.component.ts:19]: Why at a new line?

File: src/app/mini-games/savannah/components/options-input/options-input.component.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  constructor() {}
+
+  @HostListener('window:keydown', ['$event']) keyEvent(event: KeyboardEvent) {
+    if ([1, 2, 3, 4].includes(+event.key)) {
+      this.wordClicked(+event.key - 1);
+    }
+  }
+
+  ngOnInit(): void {
+    return;
```
Comment (rss-maksim) [inline src/app/mini-games/savannah/components/options-input/options-input.component.ts:23]: Why?

File: src/app/mini-games/savannah/pages/savannah-mini-game/savannah-mini-game.component.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<div class="savannah">
+  <ng-container *ngIf="game.gameState === gameState.SETTING">
```
Comment (rss-maksim) [inline src/app/mini-games/savannah/pages/savannah-mini-game/savannah-mini-game.component.html:2]: Ng Switch/Case fits better in my opinion

File: src/app/mini-games/savannah/pages/savannah-mini-game/savannah-mini-game.component.ts [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    isPaused: false,
+  };
+  constructor(private gamesService: MiniGamesHttpService) {}
+  wordsBatch$?: Subscription;
+  translationsBatch$?: Subscription;
+  translationsBatch$2?: Subscription;
+  translationsBatch$3?: Subscription;
+
+  ngOnInit(): void {
+    return;
```
Comment (rss-maksim) [inline src/app/mini-games/savannah/pages/savannah-mini-game/savannah-mini-game.component.ts:49]: unnecessary

File: src/app/mini-games/savannah/utils/utils.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  var result = [];
+  for (var i = 0; i < amount; i++) {
+    var index = Math.floor(Math.random() * array.length);
+    result.push(array[index]);
+    array.splice(index, 1);
+  }
+  return result;
+}
+
+export function sound(src: string) {
```
Comment (rss-maksim) [inline src/app/mini-games/savannah/utils/utils.ts:39]: There was already such function in utils

**Review by rss-maksim — APPROVED** [review]

File: src/app/mini-games/savannah/components/main-actions/main-actions.component.ts (hunk 4 above)
Comment (Ars6300) [reply src/app/mini-games/savannah/components/main-actions/main-actions.component.ts:19]: seems like prettier or whatever rule..

**Pull request merged**